Tools/LambdaTestTool-v2/src/Amazon.Lambda.TestTool/Services/ApiGatewayRouteConfigService.cs

```diff
@@ -0,0 +1,98 @@
+using System.Collections;
+using System.Text.Json;
+using Amazon.Lambda.TestTool.Models;
+using Amazon.Lambda.TestTool.Services.IO;
+using Microsoft.AspNetCore.Routing.Template;
+
+namespace Amazon.Lambda.TestTool.Services;
+
+/// <inheritdoc cref="IApiGatewayRouteConfigService"/>
+public class ApiGatewayRouteConfigService : IApiGatewayRouteConfigService
+{
+    private readonly List<ApiGatewayRouteConfig> _routeConfigs = new();
+
+    /// <summary>
+    /// Constructs an instance of <see cref="ApiGatewayRouteConfigService"/>
+    /// which loads and parses environment variables that match a specific prefix.
+    /// </summary>
+    /// <param name="environmentManager">A service to access environment variables.</param>
+    public ApiGatewayRouteConfigService(
+        IEnvironmentManager environmentManager)
+    {
+        var environmentVariables = environmentManager.GetEnvironmentVariables();
+
+        foreach (DictionaryEntry entry in environmentVariables)
+        {
+            var key = entry.Key.ToString();
+            if (key is null)
+                continue;
+            if (!(key.Equals(Constants.LambdaConfigEnvironmentVariablePrefix) || 
+                key.StartsWith($"{Constants.LambdaConfigEnvironmentVariablePrefix}_")))
+                continue;
+
+            var jsonValue = entry.Value?.ToString();
+            if (string.IsNullOrEmpty(jsonValue))
+                continue;
+            try
+            {
+                var config = JsonSerializer.Deserialize<ApiGatewayRouteConfig>(jsonValue);
```

Comment:
Can we check if the jsonValue starts with `[` and if so deserialize as a list instead of using exceptions for control flow?